File: Semestrul 5/Proiectarea-Detaliata-a-Sistemelor-Software/Laborator/Daria/Laborator_10/decorator/Salami.java
package Laborator.Daria.Laborator_10.decorator;

public class Salami extends PizzaDecorator {
    public Salami(Pizza pizza) {
        super(pizza);
    }

    @Override
    public String getDescription() {
        return super.getDescription() + ", Salami";
    }

    @Override
    public double getCost() {
        return super.getCost() + 5.0;
    }
}

File: Semestrul 5/Proiectarea-Detaliata-a-Sistemelor-Software/Laborator/Paul/design-patterns/decorator/src/main/kotlin/pizza/ingredients/Salami.kt
package org.paul.pizza.ingredients

import org.paul.pizza.Pizza
import org.paul.pizza.PizzaDecorator

class Salami(pizza: Pizza) : PizzaDecorator(pizza) {
    override fun getCost(): Double {
        return super.getCost() + COST
    }

    override fun getDescription(): String {
        return super.getDescription() + ", Salami"
    }

    companion object {
        private const val COST = 5.0
    }
}
